### ISSUE TITLE
Regex examples

### ISSUE THREAD

**Issue by thelawrencearm** [lithnet/ad-password-protection#60]
I can't see to figure out the correct syntax. I want to be able to ban the company name from being in the password. However whenever I try to ignore the case\use (?i)  or /i  for case insensitive it fails.  The only filter I can get to work is ^comapany([0-9]|[a-z]|[A-Z]){1,8}$   

I have tried 10 different regex testers but I always get An unexpected error occurred. regex_error(error_syntax)

I want to block any uppercase, lowercase, or cOmBonation case of the company name for appearing anywhere in the password.

any help would be appreciated.  

also, being able to the banned password list without having to turn of normalization would cool.

**Comment by ryannewington** [comment]
@thelawrencearm 

Unfortunately, c++'s version of regex doesn't allow modifiers such as for case-insensitivity.

I can't see a good reason why we would need case-sensitive regex when checking passwords, so I've changed it to be case insensitive by default.

I've also changed the regex function to use partial matching, rather than full matching as it currently is now. This was a counter-intuitive behavior.

I've attached an updated build if you'd like to test this. This hasn't yet been signed by Microsoft, so if you have LSA protected mode enabled, you'll need to wait until the signed version comes back.

[Lithnet.ActiveDirectory.PasswordProtection.zip](https://github.com/lithnet/ad-password-protection/files/5871033/Lithnet.ActiveDirectory.PasswordProtection.zip)
